gb-cpu/src/microcode/controller.rs

```diff
@@ -103,26 +94,20 @@ impl MicrocodeController {
         self.cache.push(byte)
     }
 
+    /// Push the value to the cache
+    pub fn push_u16(&mut self, value: u16) {
+        let bytes = value.to_be_bytes();
+        self.push(bytes[0]);
+        self.push(bytes[1]);
+    }
+
     /// Pop the last pushed `byte` from the cache.
     pub fn pop(&mut self) -> u8 {
         self.cache.pop().expect("not enough value stored in cache")
     }
 
-    pub fn set_dest(&mut self, ident: Ident) -> &mut Self {
-        self.target = Some(ident);
-        self
-    }
-
-    pub fn get_dest(&self) -> &Ident {
-        self.target.as_ref().expect("no dest set")
-    }
-
-    pub fn set_src(&mut self, ident: Ident) -> &mut Self {
-        self.source = Some(ident);
-        self
-    }
-
-    pub fn get_src(&mut self) -> &Ident {
-        self.source.as_ref().expect("no src set")
+    /// Pop the last u16 from the cache.
+    pub fn pop_u16(&mut self) -> u16 {
+        u16::from_be_bytes([self.pop(), self.pop()])
```

Comment:
Ça ne devrait pas être `self.cache.pop()`

Reply:
no because I expect to panic if I've less than 2 values in cache, `pop` does that check for 1 value

Reply:
But isn't `self` the controller instance itself ?

Reply:
yes but `pop` does the check to ensure the cache isn't empty when requesting an byte from it

Reply:
Oh yes, my bad